### ISSUE TITLE
Cannot run candy-machine-cli.ts : Canvas node_module is compiled against NODE_MODULE_VERSION 83

### ISSUE THREAD

**Issue by amritabithi** [metaplex-foundation/metaplex#1121]
**Describe the bug**
I am getting this build error, and have tried re-installing the canvas module, rebuilding it and downgrading and upgrading node versions, and I cannot seem to get it to ever build in a different version.

My build error output is:
/home/web/metaplex/js# ts-node /home/web/metaplex/js/packages/cli/src/candy-machine-cli.ts upload /home/web/nft-assets --env devnet --keypair /home/web/.config/solana/devnet.json
Error: The module '/home/web/metaplex/js/node_modules/canvas/build/Release/canvas.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 83. This version of Node.js requires
NODE_MODULE_VERSION 102. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Object.Module._extensions..node (node:internal/modules/cjs/loader:1179:18)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/web/metaplex/js/node_modules/canvas/lib/bindings.js:3:18)
    at Module._compile (node:internal/modules/cjs/loader:1097:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1149:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12) {
  code: 'ERR_DLOPEN_FAILED'
}

**To Reproduce**
1. Clone and build the metaplex project, then try to run the candymachine file at /metaplex/js/packages/cli/src/candy-machine-cli.ts .
2. ( tried with latest node and v14 with same results )

**Expected behavior**
I am trying to upload an NFT to the devnet using the candymachine program.

**Screenshots**
( error output is above )

**Desktop (please complete the following information):**
 - OS: Ubuntu 20
 - Browser Brave
 - Version Latest ( daily update )

**Smartphone (please complete the following information):**
n/a

**Additional context**
My current node --version shows that v17.2.0 is being used.

**Comment by amritabithi** [comment]
I was able to resolve the above issue by using "nvm install 14.17.0" to install version v14.17.0 of node and removed all others with "n rm <version>" after using "n ls" to list the installed versions, removing them all and leaving v14.17.0.

I had to repeat this for both the root user and my 'regular' user on my Ubuntu installation for the system to not complain of a conflict, and then I was able to complete my build using the "yarn install && yarn bootstrap && yarn build" commands to rebuild everything under v14.17.0.

The only problem I am at right now is the DEVNET version of the Metaplex front-end appends duplicate query string data confusing the reading of the URL parameters ( like /#/admin/#/admin/#/admin?network=devnet?network=devnet?network=devnet", as well as the top right-hand network selector dropdown always reverting to the mainnet and causing the devnet setup to time out or freeze.
